lib/ash/type/enum.ex

```diff
@@ -102,7 +126,8 @@ defmodule Ash.Type.Enum do
 
       @behaviour behaviour
 
-      @values behaviour.build_values(opts[:values])
+      @values_map behaviour.build_values_map(opts[:values])
```

Comment:
I know this was here from before you, but there is no need for `behaviour` to be a variable like this. We should remove that variable and just use `Ash.Type.Enum` everywhere. It's an unnecessary indirection. Would you mind making that change while you're extending this?

Reply:
Not a problem.  I'll make the update

Reply:
Done.  Any thoughts on replacing the manual opts validation functions with constraints?  I got close to it working, but kept running into an error that `values` wasn't provided.  Is there anything special you have to do with constraints when putting them in a macro?